our-app/src/components/common/ImmunizationListItem.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Row, Col, Image, Card } from 'react-bootstrap';
+
+class ImmunizationListItem extends React.Component {
+    state = {
+        visible: false
+    };
+    
+    toggleModal = () => {
+        this.setState(prevState => ({ visible: !prevState.visible }));
```

Comment:
Will probably want to add this to Redux instead of directly setting state here.

Reply:
Eahh, this seems like it'd be a lot to add to Redux because it is the trigger for the conditionally rendered modals to be displayed and hidden. it'd be quite the task for little pay off.. Question though, since I haven't used Redux from the ground up much, is everything normally assumed to be stored in the redux store of things like these?

Reply:
I have in the past stored everything in Redux, even simple flags. I like to have everything in there so I can see it all in one place and easily debug with dev tools. However, after doing a little research, I think it is definitely acceptable to just use component state here. The logic I found behind it is, if nothing else in the application cares about it, it is okay to do this. I agree with said logic.